src/VirtualClient/VirtualClient.Actions/ApacheBench/ApacheBenchMetricsParser.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Actions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text.RegularExpressions;
+    using Newtonsoft.Json;
+    using VirtualClient.Common.Contracts;
+    using VirtualClient.Contracts;
+
+    /// <summary>
+    /// Parser for Apache bench metrics output document.
+    /// </summary>
+    public class ApacheBenchMetricsParser : MetricsParser
+    {
+        /// <summary>
+        /// Sectionize by one or more empty lines.
+        /// </summary>
+        private static readonly Regex ApacheBenchSectionDelimiter = new (@"(\n)(\s)*(\n)", RegexOptions.ExplicitCapture);
+
+        private static readonly JsonSerializerSettings SerializerSettings = new ()
+        {
+            Converters = new List<JsonConverter>
+            {
+                new ParameterDictionaryJsonConverter()
+            }
+        };
+
+        /// <summary>
+        /// Constructor for <see cref="ApacheBenchMetricsParser"/>
+        /// </summary>
+        /// <param name="results">Raw text to parse.</param>
+        public ApacheBenchMetricsParser(string results)
+            : base(results)
+        {
+        }
+
+        /// <inheritdoc/>
+        public override IList<Metric> Parse()
+        {
+            try
+            {
+                this.Preprocess();
+                Dictionary<string, string> metricsMap = ApacheBenchMetricsParser.GetMetricsMap(this.Sections["Metrics"]);
+                return new List<Metric>()
+                {
+                    new ("Concurrency Level", Convert.ToDouble(metricsMap["Concurrency Level"]), MetricRelativity.Undefined),
+                    new ("Total requests", Convert.ToDouble(metricsMap["Total requests"]), MetricRelativity.Undefined),
+                    new ("Total time (seconds)", Convert.ToDouble(metricsMap["Total time (seconds)"]), MetricRelativity.LowerIsBetter),
```

Comment:
We can add the MetricUnits separately, instead of having it in metricName. "metrics" class has a field for metricName, metricValue, metricUnit, an overload constructor can be used here that adds unit too.